### PR TITLE
webgpu: Optimize broadcast mul/div

### PR DESCRIPTION
**Pull request by qjia7**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7437)

### CODE REVIEW

**Review by Linchenn — APPROVED** [review]
LGTM, thanks!

**Review by mattsoulanille — APPROVED** [review]
LGTM

**Comment by gyagp** [comment]
Please hold this a bit while, as Jiajia still needs some changes.

File: tfjs-backend-webgpu/src/binary_op_webgpu.ts [hunk 1]
```diff
@@ -50,46 +49,54 @@ export class BinaryOpProgram implements WebGPUProgram {
...
+        this.outputComponent = 4;
+        this.variableComponents = aDivisibleBy4 ? [4, 1] : [1, 4];
       } else {
-        this.isVec4 = false;
+        this.outputComponent = 1;
         this.type = 'plain';
-        this.workPerThread = 1;
       }
-      this.shaderKey = `binary_${this.type}_${op}`;
+      this.shaderKey = `binary_${this.type}_${op}_${this.variableComponents}`;
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/binary_op_webgpu.ts:84]: I still hope variableComponents has a meaningful value for all the paths instead of undefined.

**Pull request merged**